File: skills/decision-free-specs/languages/python/fixtures/sample.py
"""Fixture with a known inventory — used to validate inventory.py."""

import json
import os
from dataclasses import dataclass
from functools import lru_cache

__all__ = ["PAGE_SIZE", "Widget", "load_widget", "WidgetError"]

PAGE_SIZE = 50
_CACHE_DIR = "/tmp/widgets"

type WidgetId = str


class WidgetError(Exception):
    """Known error type."""


@dataclass
class Widget:
    name: str
    size: int

    def area(self) -> int:
        return self.size * self.size

    def _validate(self) -> None:
        if self.size < 0:
            raise WidgetError(self.name)


class _Internal:
    def helper(self) -> None:
        pass


@lru_cache(maxsize=128)
def load_widget(widget_id: WidgetId) -> Widget:
    payload = json.loads(os.environ.get("WIDGET", "{}"))
    return Widget(name=payload.get("name", widget_id), size=PAGE_SIZE)


def _resolve(name: str):
    return getattr(json, name)


async def _refresh() -> None:
    pass

File: skills/decision-free-specs/languages/rust/fixtures/sample.rs
//! Fixture with a known inventory — used to validate inventory.py.

use std::collections::HashMap;
use std::fmt;

pub const PAGE_SIZE: usize = 50;
static CACHE_DIR: &str = "/tmp/widgets";

pub type WidgetId = String;

#[derive(Debug, Clone)]
pub struct Widget {
    pub name: String,
    size: i64,
}

#[derive(Debug)]
pub enum WidgetError {
    Missing(WidgetId),
    Invalid { reason: String },
}

pub trait Sizable {
    fn area(&self) -> i64;
    fn label(&self) -> String {
        String::from("widget {")
    }
}

impl Sizable for Widget {
    fn area(&self) -> i64 {
        self.size * self.size
    }
}

impl fmt::Display for Widget {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        write!(f, "{}", self.name)
    }
}

#[cfg(feature = "fast")]
pub fn load_widget(id: &WidgetId) -> Result<Widget, WidgetError> {
    Ok(Widget { name: id.clone(), size: PAGE_SIZE as i64 })
}

fn build_cache() -> HashMap<WidgetId, Widget> {
    HashMap::new()
}

#[macro_export]
macro_rules! widget_of {
    ($name:expr) => {
        Widget { name: $name.to_string(), size: 1 }
    };
}

pub(crate) mod helpers {
    pub fn normalize(name: &str) -> String {
        name.trim().to_lowercase()
    }
}
